xcube/core/timecoord.py

```diff
@@ -118,17 +118,14 @@ def get_time_range_from_data(dataset: xr.Dataset, maybe_consider_metadata: bool=
     time_diff = time.diff(dim=time.dims[0]).values
     time_res = time_diff[0]
     time_regular = all([time_res - diff == np.timedelta64(0) for diff in time_diff[1:]])
+    is_cf_time = type(time.values[0]).__module__.startswith('cftime')
```

Comment:
1. `time.values` eagerly loads all time data. 
2. it is not save to use `__module__` names.

```suggestion
    is_cf_time = instanceof(time[0].values.item(), cftime.datetime)
```